File: lib-database/src/main/java/com/ustadmobile/core/db/dao/ContentCategorySchemaDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmQuery;
import com.ustadmobile.lib.database.annotation.UmRepository;
import com.ustadmobile.lib.database.annotation.UmUpdate;
import com.ustadmobile.lib.db.entities.ContentCategorySchema;
import com.ustadmobile.lib.db.sync.dao.BaseDao;
import com.ustadmobile.lib.db.sync.dao.SyncableDao;

@UmDao(readPermissionCondition = "(:accountPersonUid = :accountPersonUid)")
@UmRepository
public abstract class ContentCategorySchemaDao implements SyncableDao<ContentCategorySchema, ContentCategorySchemaDao> {

    @UmQuery("SELECT * FROM ContentCategorySchema WHERE schemaUrl = :schemaUrl")
    public abstract ContentCategorySchema findBySchemaUrl(String schemaUrl);

    @UmUpdate
    public abstract void update(ContentCategorySchema entity);

}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/ContentCategorySchemaDao.kt
package com.ustadmobile.core.db.dao

import com.ustadmobile.door.annotation.DoorDao
import androidx.room.Query
import com.ustadmobile.door.annotation.*
import com.ustadmobile.lib.db.entities.ContentCategorySchema

@DoorDao
@Repository
expect abstract class ContentCategorySchemaDao : BaseDao<ContentCategorySchema> {


    @Query("SELECT ContentCategorySchema.* FROM ContentCategorySchema")
    abstract fun publicContentCategorySchemas(): List<ContentCategorySchema>

    @Query("SELECT * FROM ContentCategorySchema WHERE schemaUrl = :schemaUrl")
    abstract fun findBySchemaUrl(schemaUrl: String): ContentCategorySchema?

}
